tigris/model.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tigris
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	api "github.com/tigrisdata/tigris-client-go/api/server/v1"
+)
+
+// User document model can have its own implementation
+// of this interface to customize metadata handling
+type model interface {
+	SetCreatedAt(t time.Time)
```

Comment:
model is not exposed but the methods are exposed. Is this intentional?

Reply:
Will export it as `MetadataSetter`